### PR TITLE
Split config loading so tests can just load a dictionary.

### PR DESCRIPTION
**Pull request by tomprince**

### CODE REVIEW

File: master/buildbot/config.py [hunk 1]
```diff
@@ -277,6 +305,7 @@ def loadConfig(cls, basedir, filename):
         if errors:
             raise errors
 
+        print type(config)
```
Comment (tardyp) [inline master/buildbot/config.py]: debug print?

File: master/buildbot/master.py [hunk 2]
```diff
@@ -228,7 +238,7 @@ def startService(self):
                 # run the master.cfg in thread, so that it can use blocking code
                 self.config = yield threads.deferToThreadPool(
```
Comment (tardyp) [inline master/buildbot/master.py]: I think we should put the deferToThread in FileLoader.

I would *love* to be able to write asynchronous config_loader

**Comment by tardyp** [comment]
I like those changes a lot!

This opens up a lot of cool stuff that I cannot measure the full scale yet.

I would like at some point to be able to say in my buildbot.tac:

configFunction="buildbot_travis.configurator"

And then configurator can fetch stuff asynchronously in the db.
For now, the way I did it is the a big hack. There is a chicken and the egg problem on the config of the db_url.

File: master/buildbot/master.py (hunk 2 above)
Comment (tomprince) [inline master/buildbot/master.py]: That seems like a reasonable change, but larger than I want to do at this point. The current change makes it easy to test code without needing to generate python code and evaluate it, just to pass an existing dictionary around.

In particular, doing that requires a fair amount of refactoring tests to support deferreds.

**Comment by tomprince** [comment]
How about
```python
from buildbot_travis import TravisLoader
...
....(..., config_loader=TravisLoader(...))
```
instead?

**Comment by tardyp** [comment]
```
from buildbot_travis import TravisLoader
...
....(..., config_loader=TravisLoader(...))
```
Yes, this can work. I can look later the details on how we can make this framework allow defereable config loader

**Comment by tardyp** [comment]
👍

**Pull request merged**